impala/_thrift_api.py

```diff
@@ -220,12 +232,24 @@ def setCustomUserAgent(self, user_agent):
   def setGetCustomHeadersFunc(self, func):
     self.__get_custom_headers_func = func
 
-  # Update HTTP headers based on the saved cookies and auth mechanism.
+  # Set the user-defined callback function which adds custom HTTP headers to outgoing
+  # messages.
+  def setGetUserDefinedCustomHeadersFunc(self, func):
```

Comment:
Is this used anywhere?
Also, couldn't the test use this function to set the callback so that passing get_user_custom_headers_func through constructors can be avoided?

Reply:
Not used and removed.
I agree that it is ugly to have to change the constructors. But the ImpalaHttpClient is hidden away so it is hard to call methods on it, especially before connect() is called. And connect() sends messages which we might want to know about. 
I welcome suggestions for better structure.